server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -467,6 +482,23 @@ private List<UploadedIndexMetadata> writeIndexMetadataParallel(ClusterState clus
             writeIndexMetadataAsync(clusterState, indexMetadata, latchedActionListener);
         }
 
+        for (IndexMetadataUploadInterceptor interceptor : indexMetadataUploadInterceptors) {
+            // We are submitting the task for async execution to ensure that we are not blocking the cluster state upload
+            String interceptorName = interceptor.getClass().getSimpleName();
+            threadpool.executor(ThreadPool.Names.GENERIC).execute(() -> {
```

Comment:
> Instead of controlling the threadpool here, the interceptor should decide which threadpool to use.

Yes, let me do this. This makes sense.

> Also upload is already async, do you need the executor at all?

Since the code is generic now, and if future someone uses this hook, it is a possibility that the code can be blocking and hurt the cluster state upload. The change intends to ensure that the cluster state upload works expected.